### DISCUSSION TITLE
get selecteded rows inside widget

### DISCUSSION THREAD

**Discussion by naseef** [filamentphp/filament#11388] [Help]
### Package

Table builder

### Package Version

v3.2.8

### How can we help you?

I have a table for which the rows are selectable using `selectable()`. I want the list of selected rows inside my Stats widget which `InteractsWithPageTable`. How can I do this?

**Comment by poxin13** [comment]
Did you ever figure this out? Been struggling with the same thing.

**Comment by e2d2-dev** [comment]
hi,

only got half of a solution. you can dispatch the records and catch them in the widget class with a listener. but i didn't found a way to dispatch without an (bulk)action.

**Reply by poxin13** [reply]
> hi,
> 
> only got half of a solution. you can dispatch the records and catch them in the widget class with a listener. but i didn't found a way to dispatch without an (bulk)action.

would you be able to share your solution for that by chance?

**Comment by e2d2-dev** [comment]
table:

```
Tables\Actions\BulkAction::make('select')
    ->dispatch(fn(Collection $records) => ['records', $records->toArray()])
```

widget:
```
#[On('records')]
public function yourFunction($records) {
    //
}
```

this needs a click, data isn't send immediately after being selected.

**Comment by e2d2-dev** [comment]
solution 2:
add filter(s) to your table, use the InteractsWithPageTable trait and the getPageTableRecords() function to get the records.
